### ISSUE TITLE
Tensorflow BROKE... for CUDA 11.0

### ISSUE THREAD

**Issue by summa-code** [tensorflow/tensorflow#42076]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 20.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary): Source
- TensorFlow version (use command below): Latest
- Python version: 3.8.2
- Bazel version (if compiling from source): 3.1.0
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version: 11.00
- GPU model and memory: RTX 2070

Python 3.8.2 (default, Jul 16 2020, 14:00:26) 
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> **import tensorflow as tf**

2020-08-05 17:27:44.548309: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcudart.so.11.0
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/xxxxxxx/.virtualenvs/tensorflowgpu/lib/python3.8/site-packages/tensorflow/__init__.py", line 433, in <module>
    _ll.load_library(_main_dir)
  File "/home/xxxxxxx/.virtualenvs/tensorflowgpu/lib/python3.8/site-packages/tensorflow/python/framework/load_library.py", line 154, in load_library
    py_tf.TF_LoadLibrary(lib)
tensorflow.python.framework.errors_impl.NotFoundError: /home/xxxxxxxx/.local/lib/python3.8/site-packages/tensorflow/core/kernels/libtfkernel_sobol_op.so: undefined symbol: _ZN10tensorflow8OpKernel11TraceStringEPNS_15OpKernelContextEb

**Comment by bhack** [comment]
Are you using CUDA 11.00? Have you compiled the source from master? 
See https://github.com/tensorflow/tensorflow/issues/42047

**Comment by summa-code** [comment]
Yes, i just did like 30 minutes ago.

**Comment by ravikyram** [comment]
@summa-code

Please try building TensorFlow with CUDA 10.1, and check if it works. Thanks!

**Comment by summa-code** [comment]
@ravikyram Are you serious with your statement ? Come on, give me a break. Tell me officially that Tensorflow is NOT compatible with CUDA 11.0. And i stated that NVIDIA took 10.X out of the view and 11.0 is the latest.

**Comment by kmhofmann** [comment]
TensorFlow admittedly has many issues, but lack of CUDA 11 support is not one of them, as far as it seems to me. The tone of @summa-code is wholly inadequate.
I compiled TF 2.3.0 today under the following conditions: Ubuntu 20.04, NVIDIA driver 450.57, CUDA 11.0.2, cuDNN v8.0.2.39, GCC 9.3.0. Absolutely no issues with test code complex enough to demonstrate GPU training functionality. Instructions used for building: https://gist.github.com/kmhofmann/e368a2ebba05f807fa1a90b3bf9a1e03.

**Comment by mihaimaruseac** [comment]
@summa-code, each release of TF is released against a specific version of CUDA. There are already issues tracking CUDA 11 support which will likely come in TF 2.4.

Regarding bug tracking and using GitHub: this is open source project and it has open source tracking. That does not mean that there cannot be a parallel internal tracking.

**Comment by theadactyl** [comment]
@summa-code please follow our [Code of Conduct](https://github.com/tensorflow/tensorflow/blob/master/CODE_OF_CONDUCT.md) and remember to treat everyone in the community (TensorFlow team members included) with respect.

**Comment by kmhofmann** [comment]
> each release of TF is released against a specific version of CUDA

This seems rather constraining, especially for early adopters. Don't you think this should be rather orthogonal, with the only requirement potentially a *minimum* version tuple of CUDA/cuDNN for a given TF version?

**Comment by mihaimaruseac** [comment]
@kmhofmann you raise a good point. There are architectural issues which make us able to only test one CUDA/cuDNN tuple. As such, we cannot provide guarantees about other values of this tuple. But, afaik, community builds can build against other versions of CUDA and report issues / send PRs.

**Comment by bhack** [comment]
This is a recurrent topic in the history of the project  (2018 or probably also older tickets) https://github.com/tensorflow/tensorflow/issues/22087#issuecomment-420402286

**Comment by summa-code** [comment]
@theadactyl and @kmhofmann , it was understandably my frustration that came out after i saw the response. And i know github is opensource but yet run by another for profit corporation. I have spent hours and hours of my time building and testing to solve a specific problem. There are some solutions that are available in MxNET for example. I wanted to give a try with Tensorflow. And for some one to come and tell me to go back to 10.1 for TF 2.4 that used to work before with 11.0 is both frustrating and thought lacked depth. We can't be simply reinstall the whole driver part in the system and rebuild tensor. What i was expecting is to give some thought for these bugs and respond accordingly giving some respect for the times we spend.

**Comment by foxik** [comment]
The original crash is in fact caused by #42978 and has been fixed by #45399.

**Comment by theGOTOguy** [comment]
I was having this problem and wanted to add my fix for any who follow, and add that there may be an issue with Tensorflow 2.4.0.

When installing any version of Tensorflow, always refer to the [chart here](https://www.tensorflow.org/install/source#linux) and check that you're using compatible versions of Tensorflow, your GPU Driver, CUDA, and CudNN.  In my case, I had tried many different approaches but none of them were clearing the undefined symbol error:  Reinstalling tensorflow, pinning packages to get the correct version of CUDA, and finally upgrading from Ubuntu 20.04 to 20.10 entirely.  I had verified using "nvidia-smi", "nvcc --version", and also the [installation guide here](https://docs.nvidia.com/deeplearning/cudnn/install-guide/index.html) to confirm that my drivers, CUDA 11, and libcudnn8 were correctly installed and were ostensibly tested and compatible with 2.4.0.  Still, I got OP's error when loading Tensorflow 2.4.0.

Finally, I tried installing tf-nightly (2.5.0.dev20210105), and it fixed the problem right away.

The fact that Tensorflow 2.4.0 showed this undefined symbol error with CUDA 11 and CudNN 8.0 whereas the nightly build worked just fine leads me to believe that there may legitimately be something broken about Tensorflow 2.4.0.
